### ISSUE TITLE
Types don't get infer on client 

### ISSUE THREAD

**Issue by enyelsequeira** [w3cj/hono-open-api-starter#4]
Hey @w3cj  I watched the video and loved the starter, I tried using it in  a monorepo like setup, but the types of the `app` router don't get infer correctly in the client which is just a vite react application, wondering if you have any workaround it? 
using somehting like this 
```// index.ts
import { Hono } from 'hono'
import authors from './authors'

const booksApp = new Hono()
  .get("/", (c) => c.json({ result: "list books" }))
  .post("/", (c) => c.json({ result: "create a book" }, 201))
  .get("/:id", (c) => c.json({ result: `get ${c.req.param("id")}` }));

const app = new Hono()

const routes = app.route('/authors', authors).route('/books', books)

export default app
export type AppType = typeof routes
``` 

this does infer the types correctly, but only infers books, routes ans not the authors, because it lives in another file. wondering if you have any idea as to why that might be? 
![image](https://github.com/user-attachments/assets/85b00009-17a7-41e5-b5f8-74c1be1044fd)
![image](https://github.com/user-attachments/assets/dc929c00-aa7a-4e9c-b418-0f8cdef6a8fb)
but then for books you can see that it does get infer correctly, since its in the same file. I also tried openAI and that just doesnt give any inference at all, any suggestions?

Thank you!

**Comment by enyelsequeira** [comment]
> Can you show the authorsApp export? You need to make sure that the exported value includes the chained methods, just like the booksApp variable.
> 
> Notice how in my tasks example, the exported router is the full chain, not just the hono app: https://github.com/w3cj/hono-open-api-starter/blob/main/src/routes/tasks/tasks.index.ts#L13

yes, of course. I am chaining them similar to the books one. Only difference is that it is in another file 

![image](https://github.com/user-attachments/assets/33f12c05-0007-4ea9-b212-b4082ae2c2c5)

**Comment by w3cj** [comment]
Ah gotcha. The calls need to be one after another like `new Hono().get().post().get()` etc. otherwise you lose the type information.

So this needs to be:

```ts
const app = new Hono()
  .get("/", ...)
  .post("/", ...)
  .get("/:id", ...)

export default app;
```

**Comment by enyelsequeira** [comment]
I was following this https://hono.dev/docs/guides/best-practices, guess it shouldn't be as the docs in here. 
this does in fact get the types correctly.  ```
//GETS TYPES CORRECTLY
const app = new Hono()
  .get("/", ...)
  .post("/", ...)
  .get("/:id", ...)

export default app;
````
 
Would you have any idea why then for example when using the setup you have, exactly the same the client is unable to get the type correctly?
![image](https://github.com/user-attachments/assets/17826a4b-96b9-4f38-82c7-c900694642f3)
![image](https://github.com/user-attachments/assets/69f8f2b6-a1cf-41cf-bcc5-265abf7e9295)

**Comment by w3cj** [comment]
Not sure, but in a typically monorepo setup, the dependent package folders are next to each other, not nested. Then instead of creating typescript aliases to the other projects, the projects list each other as dependencies and can import directly from the project package name.

For a proper setup, checkout workspaces: https://pnpm.io/workspaces

Something else to try, if you can get your server code building with typescript (with declaration files), you can import the built declaration type directly into the frontend.

**Comment by enyelsequeira** [comment]
> Not sure, but in a typical monorepo setup, the dependent package folders are next to each other, not nested. Then instead of creating typescript aliases to the other projects, the projects list each other as dependencies and can import directly from the project package name.
> 
> For a proper setup, checkout workspaces: https://pnpm.io/workspaces
> 
> Something else to try, if you can get your server code building with typescript (with declaration files), you can import the built declaration type directly into the frontend.

Hey Cj. Thank you I will try to change it to a monorepo setup either using what you shared or turborepo and see if the issue still occurs :)
